src/hotspot/share/oops/instanceKlass.hpp

```diff
@@ -285,7 +285,8 @@ class InstanceKlass: public Klass {
     _misc_invalid_inline_super                = 1 << 19, // invalid super type for an inline type
     _misc_invalid_identity_super              = 1 << 20, // invalid super type for an identity type
     _misc_has_injected_identityObject         = 1 << 21, // IdentityObject has been injected by the JVM
-    _misc_has_injected_primitiveObject        = 1 << 22  // PrimitiveObject has been injected by the JVM
+    _misc_has_injected_primitiveObject        = 1 << 22, // PrimitiveObject has been injected by the JVM
+    _misc_has_preload_attribute               = 1 << 23  // class has a Preload attribute
   };
 
```

Comment:
Is the _misc_has_preload_attribute flag needed, or can the code just check if _preload_classes equals Null?

Reply:
Strictly speaking, it is not needed. At some point, I've considered putting the _preload_classes pointer into the optional part of InstanceKlass, which would have required this flag. But finally it is declared as a regular field, so this flag can be removed.